util/codec/decimal_test.go

```diff
@@ -47,9 +47,18 @@ func (s *testDecimalSuite) TestDecimalCodec(c *C) {
 	for _, input := range inputs {
 		v := types.NewDecFromFloatForTest(input.Input)
 		datum := types.NewDatum(v)
+
 		b, err := EncodeDecimal([]byte{}, datum.GetMysqlDecimal(), datum.Length(), datum.Frac())
 		c.Assert(err, IsNil)
-		_, d, err := DecodeDecimal(b)
+		_, d, prec, frac, err := DecodeDecimal(b)
+		if datum.Length() != 0 {
+			c.Assert(prec, Equals, datum.Length())
+			c.Assert(frac, Equals, datum.Frac())
+		} else {
+			prec1, frac1 := datum.GetMysqlDecimal().PrecisionAndFrac()
```

Comment:
s/datum.GetMysqlDecimal().PrecisionAndFrac()/ v.PrecisionAndFrac()

Reply:
datum.GetMysqlDecimal().PrecisionAndFrac() is ok.